dex/src/state.rs

```diff
@@ -241,7 +241,9 @@ impl MarketStateV2 {
             .map_err(|_| DexErrorCode::InvalidMarketFlags)?;
         let required_flags =
             AccountFlag::Initialized | AccountFlag::Market | AccountFlag::Permissioned;
-        if flags != required_flags {
+        if flags != required_flags
+            && flags != (required_flags | AccountFlag::CrankAuthorityRequired)
```

Comment:
This can also be `!flags.contains(required_flags)`, though your change is more explicit.